javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,219 @@
+                }
+
+                todoArray.push(nameString2.toLowerCase());
+                console.log(`Added ${nameString2} to myToDo`);
+                //console.log(`Total items in the ToDo  ${todoArray.length}`);
+                break;
+
+            case 'Remove':    
+                if(wordString.length > 4)
+                {
+                    nameString2 = wordString;
+                    nameString2.splice(0, 1); // remove first element
+                    nameString2.splice(-1, 1); // remove last element
+                    nameString2.splice(-1, 1); // remove 2nd last element, whih is infact the last now
+                    nameString2 = nameString2.join(' '); // combine element to form the string
+                }
+                else
+                {
+                    nameString2 = wordString[1];
+                }
+
+                 // check if the item is in the list, before removing it.If exists then remove it.
+                if(todoArray.includes(nameString2.toLowerCase()))
+                {
+                    itemIndex = todoArray.indexOf(nameString2);
+                    todoArray.splice(itemIndex, 1) ; 
+                    console.log(`${nameString2} is removed from myToDo`);              
+                }
+                else 
+                {    
+                    // if not in the list, print invalid item to be removed
+                    // if item exist in the list, remove it :-) 
+                    console.log(`Invalid item to be removed`);
+                }
+                break;
+
+            
+            //----------------------------------
+            case 'What':    
+                nameString2 = wordString;
+                nameString2.splice(0, 1); // remove first element
+                nameString2 = nameString2.join(' '); // combine element to form the string
+
+                if(nameString2 == 'is my name?')
+                {
+                    if(nameArray == ' ') 
+                    {
+                        console.log('No name exist !');
+                    }
+                    else
+                    {
+                        console.log(`your name is: ${nameArray}`)  
+                    }
+                }
+                else if(nameString2 == 'is on myToDo?')
+                {
+                   console.log(`ToDo is: ${todoArray}`);  
+                }
+                else if(nameString2 == 'day is today?')
+                {
+            
+                    const date = new Date().toDateString().split(" ");
+                    console.log(`${date[2]} of ${date[1]} ${date[3]}`);
+                }
+                else
+                {
+                    // mathematic operations
+                    let operator = wordString.slice(-2,-1);  // get the maths operator
+                    let operandA = parseInt(wordString.slice(-3,-2));
+                    let operandB = parseInt(wordString.slice(-1));
```

Comment:
If you want, you can use what is called "array destructuring" here:
```js
// if wordString is ["what", "is", "2", "+", "2"]
const [ firstElement, secondElement, operandA, operator, operandB ] = wordString; 
// this will assign the values directly.
// and because you don't need the two first elements you can simply do:
const [ , , operandA, operator, operandB ] = wordString; 
```